src/EditDistPeriodicityDetector.cpp

```diff
@@ -55,60 +59,89 @@ namespace geopm
+        }
+        if (jj >= m_history_buffer_size) {
+            // This value is supposed to be INF but not so large that it gets wrapped around when
+            // a small value is added to it.
+            return std::numeric_limits<uint32_t>::max()/2;
+        }
+        if (mm <= m_record_count - m_history_buffer_size) {
+            // This value is supposed to be INF but not so large that it gets wrapped around when
+            // a small value is added to it.
+            return std::numeric_limits<uint32_t>::max()/2;
+        }
+        return m_DP[((ii % m_history_buffer_size)  * m_history_buffer_size
+                   + (jj % m_history_buffer_size)) * m_history_buffer_size
+                   + (mm % m_history_buffer_size)];
+     }
+
     void EditDistPeriodicityDetector::calc_period(void)
     {
-        int buffer_size = m_history_buffer.size();
-        if (buffer_size < 2) {
+        if (m_record_count < 2) {
             return;
         }
 
-        size_t dim_i = m_history_buffer.size();
-        size_t dim_j = dim_i + 1;
-        size_t dim_m = dim_i + 1;
-        int DD[dim_i][dim_j][dim_m];
-        for (int mm = 0; mm < buffer_size + 1; ++mm) {
-            for (int ii = 0; ii < buffer_size; ++ii) {
-                DD[ii][0][mm] = 0;
-                DD[0][ii][mm] = ii;
-            }
+        int num_recs_in_hist = m_history_buffer.size();
+
+        for (int ii = std::max({0, m_record_count - m_history_buffer_size}); ii < m_record_count; ++ii) {
+            Dset(ii, 0, m_record_count-1, 0);
+        }
+        for (int mm = std::max({0, m_record_count - m_history_buffer_size}); mm < m_record_count; ++mm) {
+            Dset(0, m_record_count - mm, mm, m_record_count - mm);
         }
 
-        for (int mm = 1; mm < buffer_size + 1; ++mm) {
-            for (int ii = 1; ii < mm; ++ii) {
-                for (int jj = 1; jj < buffer_size - mm + 2; ++jj) {
-                    int term = (get_history_value(ii) !=
-                                get_history_value(mm + jj - 1)) ?
-                               2 : 0;
-                    DD[ii][jj][mm] = std::min({DD[ii - 1][jj    ][mm] + 1,
-                                               DD[ii    ][jj - 1][mm] + 1,
-                                               DD[ii - 1][jj - 1][mm] + term});
+        for (int mm = std::max({1, m_record_count - m_history_buffer_size}); mm < m_record_count; ++mm) {
+            for (int ii = std::max({1, m_record_count - m_history_buffer_size}); ii < mm + 1; ++ii) {
+                int term = 2;
+                if ((m_record_count - (ii - 1) <= num_recs_in_hist) &&
+                    (m_history_buffer.value(num_recs_in_hist - (m_record_count - (ii - 1))) ==
+                     m_history_buffer.value(num_recs_in_hist - 1)))
+                {
+                    term = 0;
                 }
+                Dset(ii, m_record_count - mm, mm,
+                        std::min({Dget(ii - 1, m_record_count - mm    , mm) + 1,
```

Comment:
use a intermediate variable to store result of the min() computation.

Reply:
Done.